source/common/upstream/subset_lb.cc

```diff
@@ -593,6 +585,45 @@ void SubsetLoadBalancer::forEachSubset(LbSubsetMap& subsets,
   }
 }
 
+// Purge empty subsets.
+void SubsetLoadBalancer::purgeEmptySubsets(LbSubsetMap& subsets) {
+  std::vector<std::string> purge_subsets;
```

Comment:
if you want to optimize this a bit you could store a vector of iterators instead of copying the key string. might not matter